java/src/main/native/ai_onnxruntime_OrtSession.c

```diff
@@ -13,227 +13,286 @@
+  const jchar* cPath = (*jniEnv)->GetStringChars(jniEnv, modelPath, NULL);
+  size_t stringLength = (*jniEnv)->GetStringLength(jniEnv, modelPath);
+  wchar_t* newString = (wchar_t*)calloc(stringLength + 1, sizeof(wchar_t));
+  if (newString == NULL) {
+    (*jniEnv)->ReleaseStringChars(jniEnv, modelPath, cPath);
+    throwOrtException(jniEnv, 1, "Not enough memory");
+    return 0;
+  }
+  wcsncpy_s(newString, stringLength + 1, (const wchar_t*)cPath, stringLength);
+  checkOrtStatus(jniEnv, api,
+                 api->CreateSession((OrtEnv*)envHandle, newString, (OrtSessionOptions*)optsHandle, &session));
+  free(newString);
+  (*jniEnv)->ReleaseStringChars(jniEnv, modelPath, cPath);
 #else
-    const char* cPath = (*jniEnv)->GetStringUTFChars(jniEnv, modelPath, NULL);
-    checkOrtStatus(jniEnv,api,api->CreateSession((OrtEnv*)envHandle, cPath, (OrtSessionOptions*)optsHandle, &session));
-    (*jniEnv)->ReleaseStringUTFChars(jniEnv,modelPath,cPath);
+  const char* cPath = (*jniEnv)->GetStringUTFChars(jniEnv, modelPath, NULL);
+  checkOrtStatus(jniEnv, api, api->CreateSession((OrtEnv*)envHandle, cPath, (OrtSessionOptions*)optsHandle, &session));
+  (*jniEnv)->ReleaseStringUTFChars(jniEnv, modelPath, cPath);
 #endif
 
-    return (jlong) session;
+  return (jlong)session;
 }
 
 /*
  * Class:     ai_onnxruntime_OrtSession
  * Method:    createSession
  * Signature: (JJ[BJ)J
  */
-JNIEXPORT jlong JNICALL Java_ai_onnxruntime_OrtSession_createSession__JJ_3BJ
-  (JNIEnv * jniEnv, jclass jclazz, jlong apiHandle, jlong envHandle, jbyteArray jModelArray, jlong optsHandle) {
-    (void) jclazz; // Required JNI parameter not needed by functions which don't need to access their host object.
-    const OrtApi* api = (const OrtApi*) apiHandle;
-    OrtSession* session;
-
-    // Get a reference to the byte array elements
-    jbyte* modelArr = (*jniEnv)->GetByteArrayElements(jniEnv,jModelArray,NULL);
-    size_t modelLength = (*jniEnv)->GetArrayLength(jniEnv,jModelArray);
-    checkOrtStatus(jniEnv,api,api->CreateSessionFromArray((OrtEnv*)envHandle, modelArr, modelLength, (OrtSessionOptions*)optsHandle, &session));
-    // Release the C array.
-    (*jniEnv)->ReleaseByteArrayElements(jniEnv,jModelArray,modelArr,JNI_ABORT);
-
-    return (jlong) session;
-  }
+JNIEXPORT jlong JNICALL Java_ai_onnxruntime_OrtSession_createSession__JJ_3BJ(JNIEnv* jniEnv, jclass jclazz, jlong apiHandle, jlong envHandle, jbyteArray jModelArray, jlong optsHandle) {
+  (void)jclazz;  // Required JNI parameter not needed by functions which don't need to access their host object.
+  const OrtApi* api = (const OrtApi*)apiHandle;
+  OrtEnv* env = (OrtEnv*)envHandle;
+  OrtSessionOptions* opts = (OrtSessionOptions*)optsHandle;
+  OrtSession* session = NULL;
+
+  // Get a reference to the byte array elements
+  jbyte* modelArr = (*jniEnv)->GetByteArrayElements(jniEnv, jModelArray, NULL);
+  size_t modelLength = (*jniEnv)->GetArrayLength(jniEnv, jModelArray);
```

Comment:
>ize_t modelLength [](http://example.com/codeflow?start=3&length=17)

Should we check for a zero len array and throw InvalidArgument?

Reply:
Can do. I might put that check in Java though as it's easier to do. I'd assumed the create session call would throw with illegal argument anyway so I hadn't bothered.

Reply:
This would be a one place to put the check vs potentially several places in Java. I would vote to place a check here. Treat this as a Java method written in a different language.

Reply:
Fixed.